cfpb/consumerfinance.gov
# Add browserslist config to esbuild

Autoprefixer currently uses browserslist with the `default` browser set. 
To view this:
In `styles.js`, change `plugins: [ autoprefixer ]` to `plugins: [ autoprefixer.info() ]` and run `yarn styles` and see browserslist output.

This PR adds back the browserslist string to `package.json` so that both esbuild and autoprefixer are using the same browser set. 

We could do `"browserslist": "defaults",` in `package.json` if we want to, but I think as written it's a little clearer.

## Additions

- Adds `esbuild-plugin-browserslist` for feeding browserslist string in esbuild.
- Adds browserslist string to `package.json`, without IE11.


## Removals

- Unreferenced `postcss-unmq` and `require-dir`

## How to test this PR

1. In the root directory, run `npx browserslist` to see the browser targets.
2. Change browserslist string to `"browserslist": "ie >= 11",` in `package.json`
3. Re-run `npx browserslist` and see that it only outputs `ie 11`.
4. Run `export DEBUG=esbuild-plugin-browserslist`
5 Run `yarn scripts` and see it fails and esbuild info shows IE11 was targeted (which isn't enough for esbuild).
6. In `styles.js`, change `plugins: [ autoprefixer ]` to `plugins: [ autoprefixer.info() ]` 
7. Run `yarn styles` and scroll up to see browserslist output targets is only IE11.
8. Revert browserslist string in `package.json` to `> 0.5%, last 2 versions, Firefox ESR, not dead`
9. Re-run `yarn scripts` and see the targets are updated and it passes.
10. Re-run `yarn styles` and see the targets are updated.
